### ISSUE TITLE
Not possible to override file extension

### ISSUE THREAD

**Issue by torarnv** [unintended/download-organizer-chrome-extension#42]
Adding a rule with destination path `${filename}.foobar` does not seem to work. This seems to be a limitation in the Chrome API, where calling `suggest` with a filename of a known mime type will always override the file extension.

It would be good to add this limitation to the right hand side help/tips documenaton.

I was able to work around it by using another extension to modify the response headers for the same url as I'm downloading, changing the `Content-Type` header to `application/x-foobar`. Then it is possible to change the file extension.

Perhaps this extension could add a hook for `onHeadersReceived` for the downloaded URL, and change the response content type  to `application/x-download-organizer` to ensure extensions can always be changed?

**Comment by unintended** [comment]
I'm not sure we should add such behavior, because it will extend the required permissions with a heavy permission "webRequest". I will add info about to the doc.

**Comment by torarnv** [comment]
Agreed, having to extend the permissions would be unfortunate. I assume it's not possible to selectively request headers only for a specific URL, based on a "about to initiate download" callback, or one of the URL regexes the user defines?

Filed issue upstream here: https://bugs.chromium.org/p/chromium/issues/detail?id=1161162

**Comment by unintended** [comment]
@torarnv 

Hmm, I couldn't reproduce it with this rule:
`{"description":"Zip and GZip archives","enabled":true,"mime":"application/(zip|gzip)","pattern":"archives/${filename}.foobar"}`
trying to DL an archive from https://chromium.woolyss.com/

![image](https://user-images.githubusercontent.com/641973/102974489-b0533080-450f-11eb-9a78-335d43dbc5d8.png)

**Comment by unintended** [comment]
Could you share the failing rule and DL link for a file to check?

**Comment by torarnv** [comment]
The downloads from https://chromium.woolyss.com/ are served as `application/octet-stream`, not `application/zip`, but still matches the rule due to the `EXT_MIME_MAPPINGS` override. I'm guessing Chrome treats `application/octet-stream` as an "unknown" type and hence doesn't force the extension.

I was using https://bugreports.qt.io/secure/attachment/33631/hellogl_v1.zip downloaded from https://bugreports.qt.io/browse/QTBUG-31894 as my testcase.
